client/js/components/palette.js

```diff
@@ -1,20 +1,47 @@
 export default (el) => {
   const computedStyle = window.getComputedStyle(el, ':after').content.replace(/"/g, '').split(' ');
-  const palette = computedStyle.map((str, i) => {
-    if (!(i % 2)) {
-      return { name: str, hex: computedStyle[i + 1] };
-    }
-  }).filter(o => o) // Remove `undefined`s
-    .filter(o => o.hex.indexOf('#') > -1); // Remove non-hex values (e.g. 'currentColor')
-
-  palette.forEach((pair) => {
-    const descriptionEl = document.createElement('div');
-    descriptionEl.innerHTML = `
-      <div class="styleguide__hex" style="background: ${pair.hex};">
-        <span class="styleguide__color">${pair.name}: <code class="styleguide__hex-code">${pair.hex}</code></span>
-      </div>
-    `;
-
-    el.appendChild(descriptionEl.querySelector('.styleguide__hex'));
+  const palette = computedStyle.map((str) => {
+    const split1 = str.indexOf('|');
+    const split2 = str.indexOf(',');
+    const cssName = str.slice(0, split1);
+    const name = cssName.replace(/-/g, '<span class="styleguide__palette__hyphen">-</span>');
+    const hex = str.slice(split1 + 1, split2);
+    const attribution = str.slice(split2 + 1);
```

Comment:
Let's maybe have a chat about how to potentially get it directly into the template, it's an interesting challenge.